ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/KVUtilsTransactionSpec.scala

```diff
@@ -106,6 +106,114 @@ class KVUtilsTransactionSpec extends AnyWordSpec with Matchers {
         }
     }
 
+    "reject a pre-executed submission referring to a replaced key" in KVTest
+      .runTestWithSimplePackage(alice, bob, eve) { simplePackage =>
+        val seeds =
+          Stream
+            .from(0)
+            .map(i => crypto.Hash.hashPrivateKey(this.getClass.getName + i.toString))
+        for {
+          createTransaction <- runSimpleCommand(alice, seeds.head, simpleCreateCmd(simplePackage))
+          _ <- preExecuteTransaction(
+            submitter = alice,
+            transaction = createTransaction,
+            submissionSeed = seeds.head,
+          )
+
+          preExecuted <- inParallelKeepingFirst(
```

Comment:
Can you possibly make these tests easier to read, please? E.g., the initial create transaction, the seeds, most of the `preExecuted` section could be factored out leaving the core of the setup logic in the test body without the fluff.

Reply:
Unfortunately, the setup is specific to the test, and quite important. I wouldn't want to hide it away completely. I've extracted some functions to the top of each test that should help make things a little clearer.